### PR TITLE
Fix "tda-api fails to install on OSs that don't support multi-byte characters"

### PR DESCRIPTION
**Pull request by hmanfarmer**
All unit tests passed using the github CI toolchain on the hmanfarmer/tda-api fork

### 3 changes made to implement this fix:
1. In `tda/auth.py` replaced `pyreadline` with `gnureadline`. Both libraries solve the same problem, except that `pyreadline` ["PyReadline is not actively developed at the moment (early 2017), but you can still find it on PyPI and on GitHub."](https://ipython.org/pyreadline.html) and contains multibyte characters (specifically an umlaut) in file(s) read during its build process, which causes an install error on systems not configured to use a multibyte character set. `gnureadline` has neither of those problems.
2. Modified `requirements.txt` and `setup.py` so that `gnureadline` only installs and builds on Mac OSs. It is noteworthy that `gnureadline` will not install to Windows (it throws an error if you try to do so), this is on purpose because `gnureadline` was written to solve problems specific to MacOS.
3. Added a job to `python.yml` that installs tda-api to a minimal ubuntu image using a docker container. The minimal image does not come with pip or python installed, thus that is explicitly done during the test.

@alexgolec raised a concern about testing the minimal ubuntu image. I'd like to say, that this test is very much isolated from the other tests and I think it is a good test case, as there might be other interesting bugs that could pop up in the future if we add dependencies that unintentionally depend on features not typically found on minimal images.

### CODE REVIEW

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -5,6 +5,35 @@ on:
...
+
+    - name: install pip3
+      run: apt install -y python3-pip
+
+    - name: intall requirements.txt
+      run: pip3 install -r requirements.txt
+
+    - name: install from setup.py
+      run: python3 setup.py install
+
```
Comment (hmanfarmer) [inline .github/workflows/python.yml:36]: I am not running any tests on the minimal ubuntu image. Probably should run tests.

**Comment by hmanfarmer** [comment]
I just ran across the [python-prompt-toolkit library](https://python-prompt-toolkit.readthedocs.io/en/stable/) which is actively developed and very popular. We should consider using this library instead of readline. That way, we can remove the special dependency needed to support MacOS.

**Comment by alexgolec** [comment]
So I like this change, but my concern is that the minimal ubuntu testing isn't happening in the same pattern as the rest of the tests. In particular, the tests are structured as a platform-to-python version matrix, whereas the minimal ubuntu test installs in its own environment. I can see two ways of going forward, both of which are acceptable to me: 

1. Add a minimal ubuntu image as a test environment so that it works with the standard flow. I seem to remember you saying this is difficult or impossible because the ubuntu image isn't minimal by default. 
2. Just don't bother testing this particular edge case, and "test in prod" by letting people raise issues in the discord server. This is obviously more fragile, but given the small impact of this error, I think it's okay to punt on more thorough testing until more demand for this scenario is demonstrated.

**Comment by alexgolec** [comment]
Whoops, sorry, didn't mean to close the PR

**Comment by hmanfarmer** [comment]
I swapped out the 'gnureadline' implementation for 'prompt_toolkit' and I removed the minimal ubuntu testing.
